flow/core/kernel/vehicle/traci.py

```diff
@@ -1120,3 +1133,17 @@ def get_max_speed(self, veh_id, error=-1001):
     def set_max_speed(self, veh_id, max_speed):
         """See parent class."""
         self.kernel_api.vehicle.setMaxSpeed(veh_id, max_speed)
+
+    def get_soc(self, veh_id):
+        return self.__vehicles[veh_id]['SOC']
+
+    def set_soc(self, veh_id, soc):
+        """Set the lane headways of the specified vehicle."""
+        self.__vehicles[veh_id]['SOC'] = soc
+
+    def get_fuel(self, veh_id):
+        return self.__vehicles[veh_id]['fuel']
+
+    def set_fuel(self, veh_id, fuel):
+        """Set the lane headways of the specified vehicle."""
+        self.__vehicles[veh_id]['fuel'] = fuel
```

Comment:
not sure if this is the best place for these, since soc and fuel are mutually exclusive. it would be nice if (at least) they could be consolidated to something using consistent units.